packages/@okta/vuepress-site/docs/guides/sign-your-own-saml-csr/main/index.md

```diff
@@ -0,0 +1,375 @@
+
+---
+
+**Learning outcomes**
+
+* List your apps using Okta APIs.
+* Generate, sign, and publish a certificate signing request (CSR).
+* Update your app, and ISV, to use the new certificate.
+* Clone the certificate for use across multiple apps.
+
+
+**What you need**
+
+* To use your own SAML certificate, update the key credential for the affected apps or IdPs.
+
+**Sample code**
+
+n/a
+
+---
+
+## Outbound and Inbound SAML Applications
+
+Okta as a SAML Service Provider is referred to as Inbound SAML. Okta as a SAML Identity Provider (IdP) is referred to as Outbound SAML. The general procedure is the same for both, however, some of the API calls are different as described in the sections below. 
+
+> **Note:** After you update the key credential, your users can't access the SAML app until you upload the new certificate to the ISV.
+
+For information on using the Postman REST API test client for these steps, see [Get Started with the Okta REST APIs](/code/rest/).
+
+## List your apps
+
+* For Outbound SAML, use the [Apps API](/docs/reference/api/apps/#list-applications) to return a list of all apps and collect the app `id` for each app that you want to update.
+
+* For Inbound SAML, use the [IdPs API](/docs/reference/api/idps/#list-identity-providers-with-defaults) to return a list of all Identity Providers (IdP) and collect the full response for each IdP that you want to update.
+
+The following example shows a call for Outbound SAML apps.
+
+Request: `GET /api/v1/apps`
+
+Truncated Response:
+
+```json
+{
+    "id": "00000id1U3iyFqLu0g4",
+    "name": "appname",
+    "label": "Application Name",
+    "status": "ACTIVE",
+    "lastUpdated": "2015-01-24T00:09:01.000Z",
+    "created": "2014-01-06T23:42:40.000Z",
+    "accessibility": {
+      "selfService": false,
+      "errorRedirectUrl": null,
+      "loginRedirectUrl": null
+    },
+    "visibility": {
+      "autoSubmitToolbar": true,
+      "hide": {
+        "iOS": false,
+        "web": false
+      },
+      "appLinks": {
+        "login": true
+      }
+    },
+    "features": [],
+    "signOnMode": "SAML_2_0",
+    "credentials": {
+      "scheme": "EDIT_USERNAME_AND_PASSWORD",
+      "userNameTemplate": {
+        "template": "${source.login}",
+        "type": "BUILT_IN"
+      },
+      "revealPassword": true,
+      "signing": {
+        "kid": "ZcLGUsl4Xn3996YYel6KPvOxZOhNWfly5-q36CByH4o"
+      }
+    },
+    "settings": {
+      "app": {
+        "instanceType": null
+      },
+      "notifications": {
+        "vpn": {
+          "network": {
+            "connection": "DISABLED"
+          },
+          "message": null,
+          "helpUrl": null
+        }
+      }
+    }
+  }
+```
+
+## Generate a CSR
+
+* Use the [Apps API](/docs/reference/api/apps/#list-csrs-for-application) to return a list of all apps to use with Outbound SAML apps.
+* Use the [IdPs API](/docs/reference/api/idps/#list-signing-csrs-for-idp) to return a list of all IdPs to use with Inbound SAML apps.
```

Comment:
Suggest that you don't put these two items in a bullet list. Just a paragraph will do. Reason is that you don't have an intro sentence, so to start this section off with two loose bullet items is a bit awkward and doesn't read well.